python/arcticdb/version_store/_store.py

```diff
@@ -1015,34 +1015,29 @@ def batch_read_metadata(
             Dictionary of symbol mapping with the versioned items. The data attribute will be None.
         """
         _check_batch_kwargs(NativeVersionStore.batch_read_metadata, NativeVersionStore.read_metadata, kwargs)
-        meta_items = self._batch_read_meta_to_versioned_items(symbols, as_ofs, kwargs)
-        return {v.symbol: v for v in meta_items if v is not None}
+        metadatas_or_errors = self._batch_read_metadata_or_error(symbols, as_ofs, **kwargs)
+        meta_items = []
+        for metadata in metadatas_or_errors:
+            if not isinstance(metadata, DataError):
+                vitem, udm = metadata
+                if udm != None:
+                    meta = denormalize_user_metadata(udm, self._normalizer)
+                    meta_items.append(
+                        VersionedItem(
+                            symbol=vitem.symbol,
+                            library=self._library.library_path,
+                            data=None,
+                            version=vitem.version,
+                            metadata=meta,
+                            host=self.env,
+                        )
+                    )
```

Comment:
This code is now duplicated in `library.py`. What I had in mind was for `_batch_read_metadata_or_error` to do the denormalization.

Reply:
AS discussed, we will add a flag to differentiate between both logics (library and _store.py)